app/models/pipeline_run.rb

```diff
@@ -109,17 +106,32 @@ def load_results_from_s3
 
     self.pipeline_output_id = po.id
     save
+    Resque.enqueue(LoadPostprocessFromS3, id)
+
     # rm the json
     _stdout, _stderr, _status = Open3.capture3("rm -f #{downloaded_json_path} #{downloaded_stats_path}")
     # generate report
     po.generate_report
   end
 
-  def download_file(s3_path)
-    command = "mkdir -p #{local_json_path};"
-    command += "aws s3 cp #{s3_path} #{local_json_path}/;"
+  def load_postprocess_from_s3
+    byteranges_json_s3_path = "#{sample.sample_postprocess_s3_path}/#{TAXID_BYTERANGE_JSON_NAME}"
+    downloaded_byteranges_path = download_file(byteranges_json_s3_path, local_json_path)
+    return unless downloaded_byteranges_path
+    taxon_byteranges_csv_file = "#{local_json_path}/taxon_byteranges"
+    hash_array_json2csv(downloaded_byteranges_path, taxon_byteranges_csv_file, %w[taxid hit_type first_byte last_byte])
+    ` cd #{local_json_path};
+      sed -e 's/$/,#{pipeline_output.id}/' -i taxon_byteranges;
+      mysqlimport --delete --local --user=$DB_USERNAME --host=#{rds_host} --password=$DB_PASSWORD --columns=taxid,hit_type,first_byte,last_byte,pipeline_output_id --fields-terminated-by=',' idseq_#{Rails.env} taxon_byteranges;
```

Comment:
i suppose you don't want to delete all the contents in the byte ranges?

Reply:
fixed